phoenix-core/src/it/java/org/apache/phoenix/end2end/MaxLookbackIT.java

```diff
@@ -105,60 +129,52 @@ public void testTooLowSCNWithMaxLookbackAge() throws Exception {
-            assertExplainPlan(conn, sql, dataTableName, fullIndexName);
+            String sql = String.format("SELECT * FROM %s WHERE id = 'a'", dataTableName);
             int rowsPlusDeleteMarker = ROWS_POPULATED;
-            assertRawRowCount(conn, indexTable, rowsPlusDeleteMarker);
             assertRowExistsAtSCN(getUrl(), sql, beforeDeleteSCN, true);
             flush(dataTable);
-            flush(indexTable);
-            assertRawRowCount(conn, indexTable, rowsPlusDeleteMarker);
             assertRowExistsAtSCN(getUrl(), sql, beforeDeleteSCN, true);
-            long beforeFirstCompactSCN = EnvironmentEdgeManager.currentTime();
-            Thread.sleep(1);
-            majorCompact(indexTable, beforeFirstCompactSCN);
-            assertRawRowCount(conn, indexTable, rowsPlusDeleteMarker);
+            long beforeFirstCompactSCN = EnvironmentEdgeManager.currentTimeMillis();
+            injectEdge.incValue(1); //new ts for major compaction
+            majorCompact(dataTable, beforeFirstCompactSCN);
+            assertRawRowCount(conn, dataTable, rowsPlusDeleteMarker);
             assertRowExistsAtSCN(getUrl(), sql, beforeDeleteSCN, true);
             //wait for the lookback time. After this compactions should purge the deleted row
-            Thread.sleep(MAX_LOOKBACK_AGE * 1000);
-            long beforeSecondCompactSCN = org.apache.phoenix.util.EnvironmentEdgeManager.currentTimeMillis();
+            injectEdge.incValue(MAX_LOOKBACK_AGE * 1000);
+            long beforeSecondCompactSCN = EnvironmentEdgeManager.currentTimeMillis();
             String notDeletedRowSql =
-                String.format("SELECT * FROM %s WHERE val1 = 'bc'", dataTableName);
-            assertExplainPlan(conn, notDeletedRowSql, dataTableName, fullIndexName);
+                String.format("SELECT * FROM %s WHERE id = 'b'", dataTableName);
             assertRowExistsAtSCN(getUrl(), notDeletedRowSql, beforeSecondCompactSCN, true);
-            assertRawRowCount(conn, indexTable, ROWS_POPULATED);
             assertRawRowCount(conn, dataTable, ROWS_POPULATED);
             conn.createStatement().execute("upsert into " + dataTableName +
                 " values ('c', 'cd', 'cde', 'cdef')");
             conn.commit();
-            majorCompact(indexTable, beforeSecondCompactSCN);
             majorCompact(dataTable, beforeSecondCompactSCN);
             assertRawRowCount(conn, dataTable, ROWS_POPULATED);
             //deleted row should be gone, but not deleted row should still be there.
             assertRowExistsAtSCN(getUrl(), sql, beforeSecondCompactSCN, false);
             assertRowExistsAtSCN(getUrl(), notDeletedRowSql, beforeSecondCompactSCN, true);
             //1 deleted row should be gone
-            assertRawRowCount(conn, indexTable, ROWS_POPULATED);
+            assertRawRowCount(conn, dataTable, ROWS_POPULATED);
         }
     }
 
     @Test(timeout=60000L)
```

Comment:
does this value change now?

Reply:
Not sure exactly what the question means? Before it was testing the cleanup of the delete markers in an index, now it's testing the cleanup of the delete markers in a base table. The same code's getting exercised either way.

Reply:
aah! my bad, I misunderstood this.

Reply:
And now we're testing both.